File: Sistema de Control Clínico Veterinario/Sistema_Control_Clinico_Veterinario/logica/Cita.cs
using System;

namespace SistemaVeterinario.logica
{
    public class Cita
    {
        public int IdCita { get; set; }
        public DateTime Fecha { get; set; }
        public string Hora { get; set; }
        public string Motivo { get; set; }
        public string Estado { get; set; } = "Programada";
        public Cliente Cliente { get; set; }
        public Mascota Mascota { get; set; }
        public Veterinario Veterinario { get; set; }
        public Consulta Consulta { get; set; }

        public void Cancelar() => Estado = "Cancelada";
        public void Completar() => Estado = "Completada";

        public override string ToString() =>
            $"Cita #{IdCita} | {Fecha:dd/MM/yyyy} {Hora} | Estado: {Estado} | Motivo: {Motivo}";
    }
}

File: Sistema de Control Clínico Veterinario/bluej/Cita.java
import java.util.Date;

public class Cita {
    private int idCita;
    private Date fecha;
    private String hora;
    private String motivo;
    private String estado;
    private Cliente cliente;
    private Mascota mascota;
    private Veterinario veterinario;
    
    public Cita(int id, Date fecha, String hora, String motivo,
                Cliente c, Mascota m, Veterinario v) {
        this.idCita = id;
        this.fecha = fecha;
        this.hora = hora;
        this.motivo = motivo;
        this.estado = "Programada";
        this.cliente = c;
        this.mascota = m;
        this.veterinario = v;
    }
    
    public void confirmar() { this.estado = "Confirmada"; }
    public void cancelar() { this.estado = "Cancelada"; }
    public void setEstado(String estado) { this.estado = estado; }
    public String getEstado() { return estado; }
    public Date getFecha() { return fecha; }
}